lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -5300,7 +5300,9 @@ void SROA_Parameter_HLSL::flattenArgument(
     // Unwrap top-level array if primitive
     if (inputQual == DxilParamInputQual::InputPatch ||
         inputQual == DxilParamInputQual::OutputPatch ||
-        inputQual == DxilParamInputQual::InputPrimitive) {
+        inputQual == DxilParamInputQual::InputPrimitive ||
```

Comment:
are you missing a testcase for this qualifier?

Reply:
Added.